test/k8sT/Policies.go

```diff
@@ -2338,6 +2340,301 @@ var _ = SkipDescribeIf(func() bool {
+						// need to bypass this check.
+						By("Bypassing check for ingress connectivity for remote-node, which cannot be done in a two-node cluster")
+						return
+					}
+					res := kubectl.ExecInHostNetNS(context.TODO(), k8s1Name,
+						helpers.CurlFail(k8s2PodIP))
+					ExpectWithOffset(1, res).To(getMatcher(expectRemoteNodeSuccess),
+						"HTTP ingress connectivity to pod %q from remote node", k8s2PodIP)
+				}()
+
+				wg.Add(1)
+				go func() {
+					defer GinkgoRecover()
+					defer wg.Done()
+					By("Checking ingress connectivity from k8s1 pod to k8s2 pod")
+					res := kubectl.ExecPodCmd(testNamespace, k8s1PodName, helpers.CurlFail(k8s2PodIP))
+					ExpectWithOffset(1, res).To(getMatcher(expectPodSuccess),
+						"HTTP ingress connectivity to pod %q from pod %q", k8s2PodIP, k8s1PodIP)
+				}()
+
+				if helpers.ExistNodeWithoutCilium() {
+					wg.Add(1)
+					go func() {
+						defer GinkgoRecover()
+						defer wg.Done()
+						By("Checking ingress connectivity from world to k8s1 pod")
+						By("Adding a static route to %s on the %s node (outside)", k8s1PodIP, outsideNodeName)
+						res := kubectl.AddIPRoute(outsideNodeName, k8s1PodIP, k8s1IP, true)
+						Expect(res).To(getMatcher(true))
+
+						if expectWorldSuccess {
+							testCurlFromOutside(kubectl, &nodesInfo{
+								outsideNodeName: outsideNodeName,
+							}, k8s1PodIP, 1, false)
+						} else {
+							testCurlFailFromOutside(kubectl, &nodesInfo{
+								outsideNodeName: outsideNodeName,
+							}, k8s1PodIP, 1)
+						}
+					}()
+				}
+				wg.Wait()
+			}
+
+			It("Allows connection to KubeAPIServer", func() {
+				installDefaultDenyIngressPolicy(
+					kubectl,
+					testNamespace,
+					validateConnectivity,
+				)
+				installDefaultDenyEgressPolicy(
+					kubectl,
+					testNamespace,
+					validateConnectivity,
+				)
+
+				By("Verifying KubeAPIServer connectivity is not yet allowed")
+				Expect(
+					kubectl.ExecPodCmd(
+						testNamespace, k8s2PodName, helpers.CurlWithHTTPCode(
+							"https://%s %s",
+							kubeAPIServerService.Spec.ClusterIP,
+							"--insecure", // kube-apiserver needs cert, skip verification
+						),
+					),
+				).To(getMatcher(false),
+					"HTTP egress connectivity should have been denied to pod %q to kube-apiserver %q",
+					k8s2PodName, kubeAPIServerService.Spec.ClusterIP,
+				)
+
+				By("Installing toEntities KubeAPIServer")
+				importPolicy(
+					kubectl,
+					testNamespace,
+					cnpToEntitiesKubeAPIServer,
+					"to-entities-kube-apiserver",
+				)
+
+				By("Verifying policy correctness")
+				validateConnectivity(
+					true,  /*HostConnectivityAllow*/
+					false, /*RemoteNodeConnectivityDeny*/
+					false, /*PodConnectivityDeny*/
+					false, /*WorldConnectivityDeny*/
+				)
+
+				By("Verifying KubeAPIServer connectivity")
+				// HTTP level error means connection succeeded. We expect to
+				// get back 403 because we purposefully didn't provide the auth
+				// token to fully talk to the kube-apiserver.
```

Comment:
Maybe add that 403 is a sign of success in this test because there is no egress HTTP policy in place. If there were, we would get 403 for anything denied by the HTTP policy!